src/mocks.ts

```diff
@@ -85,6 +91,18 @@ export class PlatformMock {
   }
 }
 
+export class PlatformMockIsCore {
+  public is(platform: string): Boolean {
+    return platform === 'core'
+  }
+}
+
+export class PlatformMockIsAndroid {
```

Comment:
Instead of creating both of these mocks, you could modify `PlatformMock` to have an attribute `platform`, add the method `is(platform: string)` and compare the class variable of the class with the parameter.

That way, you don't have to create a new instance of apiUrl with the appropriate mock for every test. You can instead assign the platform you want like this: `apiUrl.platform.platform = 'android'` and initialize `apiUrl` in `beforeEach()`.

You can look at other mocks that have a `resolve` variable if this explanation is not clear.